dokka-subprojects/plugin-base-frontend/webpack.config.js

```diff
@@ -2,82 +2,101 @@
-  },
-  module: {
-    rules: [
-      ...ringUiWebpackConfig.config.module.rules,
-      {
-        test: /\.s[ac]ss$/i,
-        use: [
-          MiniCssExtractPlugin.loader,
-          'css-loader',
-          'sass-loader',
-        ],
-        include: componentsPath,
-        exclude: ringUiWebpackConfig.componentsPath,
+const webpackConfig = (params, argv) => {
+  const mode = argv.mode || 'development';
+  return {
+    entry: `${componentsPath}/root.tsx`,
+    resolve: {
+      mainFields: ['module', 'browser', 'main'],
+      extensions: ['.tsx', '.ts', '.js', '.svg'],
+      alias: {
+        react: resolve('./node_modules/react'),
+        'react-dom': resolve('./node_modules/react-dom'),
+        '@jetbrains/ring-ui': resolve('./node_modules/@jetbrains/ring-ui'),
       },
-      {
-        test: /\.tsx?$/,
-        use: [
-          {
-            loader: 'ts-loader',
-            options: {
-              transpileOnly: true
-            }
-          }
-        ]
-      },
-      {
-        test: /\.svg$/,
-        loader: require.resolve('svg-inline-loader'),
-        options: {removeSVGTagAttrs: false},
-        include: [require('@jetbrains/icons')]
-      }
-    ]
-  },
-  plugins: [
-    new MiniCssExtractPlugin({
-      // Options similar to the same options in webpackOptions.output
-      // both options are optional
-      filename: '[name].css',
-      chunkFilename: '[id].css',
-    }),
-  ],
-  optimization: {
+    },
+    module: {
+      rules: [
+        ...ringUiWebpackConfig.config.module.rules,
+        {
+          test: /\.s[ac]ss$/i,
+          use: [MiniCssExtractPlugin.loader, 'css-loader', 'sass-loader'],
+          include: componentsPath,
+          exclude: ringUiWebpackConfig.componentsPath,
+        },
+        {
+          test: /\.tsx?$/,
+          use: [
+            {
+              loader: 'ts-loader',
+              options: {
+                transpileOnly: true,
+              },
+            },
+          ],
+        },
+        {
+          test: /\.svg$/,
+          loader: require.resolve('svg-inline-loader'),
+          options: { removeSVGTagAttrs: false },
+          include: [require('@jetbrains/icons')],
+        },
+      ],
+    },
+    plugins: [
+      new MiniCssExtractPlugin({
+        // Options similar to the same options in webpackOptions.output
+        // both options are optional
+        filename: '[name].css',
+        chunkFilename: '[id].css',
+      }),
+      ...(mode === 'development'
+        ? [
+            new WebpackShellPluginNext({
+              onDoneWatch: {
+                scripts: [
+                  'echo "[search] Done rebuild, coping files to dokka-integration-tests"',
+                  'cp -r ./dist/main.js ../../dokka-integration-tests/gradle/build/ui-showcase-result/scripts',
+                  'cp -r ./dist/main.css ../../dokka-integration-tests/gradle/build/ui-showcase-result/styles',
```

Comment:
Recursive flag doesn't have any affect here, let's remove it.
```suggestion
                  'cp ./dist/main.js ../../dokka-integration-tests/gradle/build/ui-showcase-result/scripts',
                  'cp ./dist/main.css ../../dokka-integration-tests/gradle/build/ui-showcase-result/styles',
```